keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/diff/DualNumber.java

```diff
@@ -350,4 +351,84 @@ private DoubleTensor pluckFromPartial(DoubleTensor from, int... indices) {
             .reshape(pluckShape);
     }
 
+    public DualNumber ifThenElse(BooleanTensor predicate, DualNumber els) {
+        if (predicate.allTrue()) {
+            return new DualNumber(this.value, this.getPartialDerivatives());
+        } else if (predicate.allFalse()) {
+            return new DualNumber(els.value, els.getPartialDerivatives());
+        } else {
+            return pluckFromMixedPredicate(predicate, this, els);
+        }
+    }
+
+    private DualNumber pluckFromMixedPredicate(BooleanTensor predicate, DualNumber thn, DualNumber els) {
+        int[] thenShape = thn.value.getShape();
+        double[] flatPredicate = predicate.asFlatDoubleArray();
+        Map<Long, List<DoubleTensor>> ifAndElseDualNumbers = new HashMap<>();
```

Comment:
this map gets passed through several private functions, changing name each time (`ifAndElseDualNumbers` --> `partials` --> `toBeConcatted`). 

There's a few things you could consider: 
- make it a field
- have each of your private methods create and return a map
- keep the logic the same but make the names consistent

Reply:
Good spot, renamed them all to match